ui/src/app/subnets/views/SubnetDetails/StaticRoutes/AddStaticRouteForm/AddStaticRouteForm.tsx

```diff
@@ -0,0 +1,146 @@
+  metric: Yup.number().required("Metric is required"),
+});
+
+const getDestinationsForSource = (subnets: Subnet[], source: Subnet | null) => {
+  const filtered: Subnet[] = [];
+  if (source) {
+    subnets.forEach((subnet) => {
+      if (subnet.id !== source.id && subnet.version === source.version) {
+        filtered.push(subnet);
+      }
+    });
+  }
+
+  return filtered;
+};
+
+export type Props = {
+  subnetId: Subnet[SubnetMeta.PK];
+  handleDismiss: () => void;
+};
+const AddStaticRouteForm = ({
+  subnetId,
+  handleDismiss,
+}: Props): JSX.Element | null => {
+  const staticRouteErrors = useSelector(staticRouteSelectors.errors);
+  const saving = useSelector(staticRouteSelectors.saving);
+  const saved = useSelector(staticRouteSelectors.saved);
+  const dispatch = useDispatch();
+  const staticRoutesLoading = useSelector(staticRouteSelectors.loading);
+  const subnetsLoading = useSelector(subnetSelectors.loading);
+  const staticRoutesLoaded = useSelector(staticRouteSelectors.loaded);
+  const subnetsLoaded = useSelector(subnetSelectors.loaded);
+  const loading = staticRoutesLoading || subnetsLoading;
+  const subnets = useSelector(subnetSelectors.all);
+  const source = useSelector((state: RootState) =>
+    subnetSelectors.getById(state, subnetId)
+  );
+  const destinationSubnets = getDestinationsForSource(subnets, source);
+
+  useEffect(() => {
+    if (!staticRoutesLoaded) dispatch(staticRouteActions.fetch());
+    if (!subnetsLoaded) dispatch(subnetActions.fetch());
+  }, [dispatch, staticRoutesLoaded, subnetsLoaded]);
+
+  if (loading) {
+    return <Spinner text="Loading..." />;
+  }
+
+  return (
+    <FormikForm<AddStaticRouteValues>
+      aria-label={AddStaticRouteFormLabels.AddStaticRoute}
+      cleanup={staticRouteActions.cleanup}
+      errors={staticRouteErrors}
+      initialValues={{
+        source: subnetId,
+        gateway_ip: "",
+        destination: "",
+        metric: "0",
+      }}
+      onSaveAnalytics={{
+        action: AddStaticRouteFormLabels.Save,
+        category: "Subnet",
+        label: AddStaticRouteFormLabels.AddStaticRoute,
+      }}
+      onSubmit={({ gateway_ip, destination, metric }) => {
+        dispatch(staticRouteActions.cleanup());
+        dispatch(
+          staticRouteActions.create({
+            source: subnetId,
+            gateway_ip,
+            destination: toFormikNumber(destination) as number,
+            metric: toFormikNumber(metric),
+          })
+        );
+      }}
+      onSuccess={() => {
+        handleDismiss();
+      }}
+      resetOnSave
+      saving={saving}
+      saved={saved}
+      submitLabel={AddStaticRouteFormLabels.Save}
+      onCancel={handleDismiss}
+      validationSchema={addStaticRouteSchema}
+    >
+      <Row>
+        <Col size={4}>
+          <FormikField label={Labels.GatewayIp} name="gateway_ip" type="text" />
+        </Col>
+        <Col size={4}>
+          <FormikField
```

Comment:
It looks like this could probably use the `SubnetSelect` component which has a `filterFunction` to which the `getDestinationsForSource` function could be passed.